lib/chef/knife/ec2_server_create.rb

```diff
@@ -1108,15 +1108,22 @@ def wait_for_direct_sshd(hostname, ssh_port)
       end
 
       def ssh_connect_host
-        @ssh_connect_host ||= if config[:server_connect_attribute]
-                                server.send(config[:server_connect_attribute])
-                              else
-                                if vpc_mode? && !config[:associate_public_ip]
-                                  server.private_ip_address
-                                else
-                                  server.dns_name || server.public_ip_address
-                                end
-                              end
+        connect_attribute = ""
+        if config[:server_connect_attribute]
+          @ssh_connect_host ||= server.send(config[:server_connect_attribute])
+          connect_attribute = config[:server_connect_attribute]
+        else
+          if vpc_mode? && !config[:associate_public_ip]
```

Comment:
Hi,

I've never seen something like that before, it's fantastic! You should take a look  <http://nonaguquo.customercareofamerica.com/oibxxa>

All best, hebgb@blueyonder.co.uk